sc/social/like/browser/viewlets.py

```diff
@@ -57,15 +65,33 @@ def update(self):
         return
 
 
-class SocialMetadataViewlet(BaseLikeViewlet):
-    """Viewlet used to insert metadata into page header
+class SocialMetadataViewlet(ViewletBase):
```

Comment:
Shouldn't you still inherit `BaseLikeViewlet` before removing it completely if your plan is backwards compatibility? If you're no inheriting it, what's the motivation of keeping it above and adding a `TODO` to remove instead of removing it already?

Reply:
it's more complicated: in fact we broke the plugin feature in release 2.5 after migrating everything to plone.app.registry. I was thinking about that yesterday when I was reviewing the work done. We didn't want to change too many things in this PR but it's inevitable because the code is way too complex.

I we really want to maintain the plugin extensibility we need to redesign it from scratch to use only on utility and one view per plugin, and using the extension mechanism already present in z3c.form. unfortunately, we have not enough heads and hands to do so in the way we may want.

we'll need to rework the whole PR.